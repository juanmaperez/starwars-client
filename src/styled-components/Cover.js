import styled from 'styled-components'

export const CoverWrapper = styled.div`
    padding: 50px;
    box-sizing: border-box;
    .welcome {
      width: 100%;
      text-align: center;
      font-size: 40px;
      font-weight: 600;
      font-family: Helvetica, sans-serif;
    }
    .leia {
      width: 100px;
      margin: auto;
      transform: scale(1.5);

      .head {
        position: relative;
        width: 87px;
        height: 96px;
        border-radius: 35%;
        background: #f6c3a4;

        &::before {
          position: absolute;
          width: 19px;
          height: 60px;
          top: 14px;
          left: -19px;
          content: "";
          background:  #473e37;
          border-radius: 9px;
        }

        &::after {
          position: absolute;
          width: 19px;
          height: 60px;
          top: 15px;
          right: -19px;
          content: "";
          background: #473e37;
          border-radius:9px;
        }

        .fringe {
          position: absolute;
          width: 50%;
          height: 39px;
          right: 0;
          background: #473e37;
          border-top-left-radius: 78%;
          border-bottom-right-radius: 70%;
          transform: scaleX(-1);

          &:first-child {
            transform: scaleX(1);
            left: 0;
          }

        }

        .eye {
          position: absolute;
          width: 7px;
          height: 8px;
          top: 52px;
          right: 20px;
          background: #222;
          border-radius: 50%;

          &:first-child {
            left: 20px;
          }

          &::before {
            content: "";
            position: absolute;
            top: -5px;
            left: -5px;
            right: 0;
            height: 6px;
            width: 18px;
            margin: 0 auto;
            border-top: 4px solid #222;
            border-radius: 100%;
          }
          
          &::after {
            content: "";
            position: absolute;
            top: 2px;
            left: 1px;
            width: 2px;
            height: 2px;
            background: #fff;
            border-radius: 50%;
          }

        }
        
        .blush {
          position: absolute;
          width: 1px;
          height: 1px;
          top: 72px;
          right: 13px;
          background: #ffa8a8;
          border-radius: 50%;
          opacity: 0.8;
          box-shadow: 0px 0px 15px 8px #ffa8a8;
          
          &:first-child {
            left: 13px;
          }  
        }
        
        .mouth {
          position: absolute;
          top: 73px;
          left: 0;
          right: 0;
          height: 8px;
          width: 19px;
          margin: 0 auto;
          border-bottom: 2px solid #222;
          border-radius: 100%;
        }
      }
    }
`

