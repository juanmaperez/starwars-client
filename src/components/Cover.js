import React from 'react';
import { CoverWrapper } from './../styled-components/Cover'

export default () => (
  <CoverWrapper>
    <div className="leia">
      <div className="head">
        <div className="fringe"></div>
        <div className="fringe"></div>
        <div>
          <div className="eye"></div>
          <div className="eye"></div>
        </div>
        <div>
          <div className="blush"></div>
          <div className="blush"></div>
        </div>      
        <div className="mouth"></div>
      </div>
    </div>
    <div className="welcome">Welcome to Start Wars</div> 
  </CoverWrapper>
)