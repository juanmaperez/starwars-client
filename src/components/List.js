import React from 'react'
import styled from 'styled-components'
import { TabContent, TabPane } from 'reactstrap'
import Cover from './Cover'
import Checkbox from './Checkbox';

const Resource = styled.div`
  border-radius: 100px;
  box-shadow: 0px 2px 5px 0px rgba(0,0,0,0.09);
  padding: 10px 20px;
  margin: 10px;
  position: relative;
  cursor: pointer;
  .description {
    position: absolute;
    left: 0px;
    top: 50px;
    width: 300px;
    background: #fff;
    box-shadow: 0px 2px 5px 0px rgba(0,0,0,0.09);
    border-radius: 10px;
    padding: 10px;
    display: none;
    z-index: 10;
    span {
      display: block;
      width: 100%;
    }
  }
  &:hover {
    .description {
      display block;
    }
  }
`
const Flex = styled.div`
  display: flex;
  flex-wrap: wrap;
  margin: 30px 0;
`

export default ({ resources, tab, keys }) => {
  const { results } = resources;
  return results
    ? <TabContent className={tab + ' table'} activeTab={tab}>
        {keys.map(k => (
            <TabPane
              key={k}
              tabId={k}
            >
              <Flex>
              { results.map((item, i) => 
                <Resource key={i}>
                  <span>{ item.name || item.title }</span>
                  <div className="description">
                    {Object.keys(item).map(k => <span key={k}><strong>{ k.replace(/_/g, ' ') }: </strong>{ item[k] }</span>)}
                  </div>
                </Resource>)}
              </Flex>
            </TabPane>
        ))}
      <Checkbox category={tab}/>
      </TabContent>
    : <Cover />
}