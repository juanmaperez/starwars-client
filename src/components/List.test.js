import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'
import List from './List'
import Cover from './Cover'
import { TabContent } from 'reactstrap';

configure({ adapter: new Adapter() })

describe('<List />', () => {

  it('List snapshot matches', () => {
    const wrapper = shallow(<List resources={{ results: []}} keys={['item', 'item2']}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('If results, load the content', () => {
    const wrapper = shallow(<List resources={{ results: []}} keys={['item', 'item2']}/>)
    expect(wrapper.find(TabContent).length).toBe(1)
  })

  it('If not results, load the cover', () => {
    const wrapper = shallow(<List resources={{ results: null}} keys={['item', 'item2']}/>)
    expect(wrapper.find(Cover).length).toBe(1)
  })

})