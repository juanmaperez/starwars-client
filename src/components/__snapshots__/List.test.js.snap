// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> List snapshot matches 1`] = `
<TabContent
  className="undefined table"
  tag="div"
>
  <TabPane
    key="item"
    tabId="item"
    tag="div"
  >
    <styled.div />
  </TabPane>
  <TabPane
    key="item2"
    tabId="item2"
    tag="div"
  >
    <styled.div />
  </TabPane>
  <_default />
</TabContent>
`;
